src/crimson/os/cyan_store.cc

```diff
@@ -287,19 +313,78 @@ seastar::future<> CyanStore::do_transaction(CollectionRef ch,
         r = _setattrs(cid, oid, to_set);
       }
       break;
-    case Transaction::OP_MKCOLL:
+      case Transaction::OP_MKCOLL:
       {
         coll_t cid = i.get_cid(op->cid);
         r = _create_collection(cid, op->split_bits);
       }
       break;
-    default:
-      logger().error("bad op {}", static_cast<unsigned>(op->op));
-      abort();
-    }
-    if (r < 0) {
-      abort();
+      case Transaction::OP_OMAP_SETKEYS:
+      {
+        coll_t cid = i.get_cid(op->cid);
+        ghobject_t oid = i.get_oid(op->oid);
+        map<string, bufferlist> aset;
+        i.decode_attrset(aset);
+        r = _omap_set_values(cid, oid, aset);
```

Comment:
Possibly there is a chance for `std::moving` the `aset` to `_omap_set_values()`.

Reply:
_omap_set_values is taking a const &, so probably not important.

Reply:
`_omap_set_values()` consumes the container to store its pairs in `Object::omap`, so it could take just `std::map<std::string, ceph::bufferlist> aset`. Still, I agree it can be unimportant – no evidence from e.g. profiling we need to optimize this.

Reply:
Ok, I'll switch it to consume the container.